### ISSUE TITLE
Create placeholder content when running standard install profile

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#519]
Instead of the unhelpful "no content created yet" message, how about a default "Your first article" node like WP does?

---
~~PR from @docwilmot at https://github.com/backdrop/backdrop/pull/1566~~
PR from @jenlampton at https://github.com/backdrop/backdrop/pull/1578

**Comment by cellear** [comment]
This seems related to this issue I opened last year.

https://github.com/backdrop/backdrop-issues/issues/89

**Comment by docwilmot** [comment]
> Posted by nedjo on August 19, 2006 at 6:51pm

Wow. I hope we're not still discussing thos 8 years from now!
I'm thinking in the default install profile a simple:
```
node>title="Your first article!"
node>body="This is your first article, edit it or delete it."
node>save.
```

Thats it. Obvious advantages over "Duh you dont have a post yet!"

**Comment by serundeputy** [comment]
Does it make sense to put that in `core/modules/system/system.install`
something like:
```php
function system_update_1036() {
  $node = array(
```
or am i way off base here?

**Comment by quicksketch** [comment]
We'd probably put it in `standard.install` (in the Standard profile), so other profiles or distributions wouldn't need our sample node or could make their own. We can do it right after the "article" content type is created.

**Comment by quicksketch** [comment]
Nice one @serundeputy. PR at https://github.com/backdrop/backdrop/pull/713. We'll need to wait on this until the 1.1.x branch exists at least. In the mean time, anyone else have input? The PR does exactly what @docwilmot described.

**Comment by serundeputy** [comment]
could be a 1.1 milestone?  (crickets)

**Comment by quicksketch** [comment]
Sure! No reason why not. I don't think this would be considered an API change.

The PR still needs a small amount of work, and should be squashed together to make a single commit.

**Comment by serundeputy** [comment]
updated and `rebased`
travis testing now.

**Comment by quicksketch** [comment]
Thanks @serundeputy. It looks like we may have a lot of tests that assume either the default front page content or a situation where the first node created is "node ID 1" (which it won't be after this change). We can probably fix most of these tests by adding:

```php
protected $profile = 'minimal';
```

At the top of some of these test classes. That will switch the installation profile from "standard" to "minimal", which doesn't include this change. The minimal install profile also contains fewer modules and runs faster anyway, so we should be using it whenever we can in tests. If switching to the minimal profile breaks more than it fixes, we'll just need to update the tests to work with the new front page node.

**Comment by serundeputy** [comment]
sounds good I'll give it a go tonight.

**Comment by serundeputy** [comment]
with 
```php
protected $profile = 'minimal';
```
![screen shot 2015-03-31 at 8 35 48 pm](https://cloud.githubusercontent.com/assets/6763671/6932329/c1d3e5d6-d7e5-11e4-8483-29284f6ae8b8.png)

**Comment by serundeputy** [comment]
if i put the same in `views_glossary.test`:

```php
/**
 * Tests glossary view ( summary of arguments ).
 */
class ViewsGlossaryTestCase extends ViewsSqlTest {
  protected $profile = 'minimal';
```
they pass.  @quicksketch what do you think?

**Comment by quicksketch** [comment]
That's great. The more tests we convert to using the minimal (or even better the "testing") profile, the faster our tests will run. Any test that can be converted to use a lighter installation profile we should switch.

**Comment by serundeputy** [comment]
updated
rebased
pushed
travis testing.

should we open a ticket to convert tests to 'testing' profiles; perhaps per module to make more manageable?

**Comment by serundeputy** [comment]
good novice tickets to get people involved?

**Comment by quicksketch** [comment]
> good novice tickets to get people involved?

It depends... sometimes I feel like our "novice" issues are still pretty difficult. SimpleTest is not a very trivial tool to understand. Though once you understand the UI for it, it can be pretty straightforward. We could mark them novice if we write up a details set of instructions.

**Comment by serundeputy** [comment]
travis test failed; running through tests on local updating to use `minimal` install profile.

**Comment by serundeputy** [comment]
so it looks like tests that depend on logging in a user with `administer views` permissions are failing. For example `views_ui.test` line 18:
```php
$views_admin = $this->backdropCreateUser(array('administer views', 'administer blocks', 'bypass node access', 'access user profiles', 'view revisions'));
```

**Comment by docwilmot** [comment]
Should we also add this node as an entry in the Main Menu? For similar reasons, it lets the new Backdropper know that the menu exists, and leads them there if they want to learn more about menus, or to delete or modify it.

**Comment by serundeputy** [comment]
sounds good to me.  @quicksketch what do you think?

**Comment by quicksketch** [comment]
What should the menu item say, "About" or "Example"? node/1 is usually an about page for me. It may be less frustrating for a new user to just modify the content of the node to be an about page than find the menu settings on the node form.

**Comment by docwilmot** [comment]
I think the menu entry should be the node title.

**Comment by jenlampton** [comment]
@cellear mentioned over in another issue that WordPress uses the following as default content:

> This is an example of a page. Unlike posts, which are displayed on your blog’s front page in the order they’re published, pages are better suited for more timeless content that you want to be easily accessible, like your About or Contact information. Click the Edit link to make changes to this page or add another page.

My thoughts....  Wouldn't this information also be useful to a second or third person, even after this first piece of content has been updated, and the information removed? 

WordPress assumes that the person building the site will be the person creating / editing content. So the message in a default node is likely to be seen by the 'correct' person. In Drupal and Backdrop the person building the site is the person least likely to need instructions on how to edit content, and the person most likely to delete or edit away the valuable information before it gets to the eyes of those who need it.

We need to make things easier for beginners (and the first time experience is important!). But hopefully we can make things easier for all site users, not just the person (more likely the guru) who built the site.

**Comment by quicksketch** [comment]
I've closed the in-progress PR at https://github.com/backdrop/backdrop/pull/713 until it can be updated to fix the remaining test failures. We're making progress here!

> My thoughts.... Wouldn't this information also be useful to a second or third person, even after this first piece of content has been updated, and the information removed? 

That may true, and there's no way we can guarantee the first user won't modify the node. It sounds like this would be describing an entire introduction experience, not just providing a default node in the standard profile. I think we should move forward with this approach for the simplicity and helpfulness it provides. A full first-time user experience is probably more inline with Tour module (#272), which has a much larger set of challenges.

**Comment by quicksketch** [comment]
I'm still up for including this in 1.1.0 if we can get it done in the next couple of days. Any takers?

**Comment by serundeputy** [comment]
@quicksketch 
one of the failures is in `bootstrap.test` for `assertTitle` (line 209) looking for `class="title"` which exists when there is no content on the front page, but does not when there is.

Since this puts content on the front page `class="title"` DNE, should we just remove this line from the test?

**Comment by docwilmot** [comment]
`assertText` instead for the "Your first article!" string?

**Comment by klonos** [comment]
Almost forgot about this one.

**Comment by serundeputy** [comment]
haha; one near and dear to my heart 😄 , but I don't think it is going into 1.4; moving to 1.5.

**Comment by jenlampton** [comment]
No progress made on this since last release, officially bumping from 1.5 to 1.x-future.

**Comment by docwilmot** [comment]
Update to this, assigning me. I'm suggesting **two** default nodes:
- A page with a menu link. Allows the user to just modify to an "About us" page as per https://github.com/backdrop/backdrop-issues/issues/519#issuecomment-90764306 above, and also gives an opportunity to introduce the menu system.
- A default post that appears on the ~~front~~ home page.

**Comment by jenlampton** [comment]
Reassigning to 1.5 since it looks like it's got a shot! :)

**Comment by quicksketch** [comment]
Get 'em @docwilmot!

**Comment by jenlampton** [comment]
@docwilmot got it so, so very close. I've fixed the last three tests and they are testing now at https://github.com/backdrop/backdrop/pull/1578

**Comment by quicksketch** [comment]
Looking good! Tests are passing but there are a few issues:

- The page node isn't added to the primary menu as a link.
- For some reason the aliases aren't working, they use the defaults of `post/first-post` and `page/about-page`. It should be `posts/first-post` and `about-page`.
- IMO, "About page" should be retitled to just "About", I get that we're including the content type in the title, but it seems so much more awkward for the About page (which probably won't be deleted)  than it does for the first post (which probably will).

**Comment by jenlampton** [comment]
> For some reason the aliases aren't working,

Looks like the nodes were created before the aliases were set. I'll move them to below.

> "About page" should be retitled to just "About"

Agree, and done.

> The page node isn't added to the primary menu as a link.

That's because the path wasn't matching up. Fixed up now, and testing.

**Comment by quicksketch** [comment]
https://github.com/backdrop/backdrop/pull/1578 has been merged into 1.x for 1.5.0. Yay! This was our last item for 1.5.0!
